### PR TITLE
Feature/photoatomic 2

### PR DESCRIPTION
**Pull request by whaeck**
Finishing the photoatomic data table:
- implementing the optional data block
- additional testing

### CODE REVIEW

File: python/src/PhotoatomicTable.python.cpp [hunk 1]
```diff
@@ -50,7 +59,10 @@ void wrapPhotoAtomicTable( python::module& module, python::module& ) {
     "    eszg      the principal cross section block\n"
     "    jinc      the incoherent scattering function block\n"
     "    jcoh      the coherent form factor block\n"
-    "    lhnm      the heating numbers block"
+    "    lhnm      the heating numbers block\n"
+    "    lhnm      the fluorescence data block\n"
```
Comment (cjosey) [inline python/src/PhotoatomicTable.python.cpp:63]: Comment is incorrect.

File: python/src/block/PhotoatomicComptonProfileBlock.python.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    "Initialise the block\n\n"
+    "Arguments:\n"
+    "    self        the block\n"
+    "    profiles    the compton profile data"
+  )
+  .def_property_readonly(
+
+    "NSH",
+    &Block::NSH,
+    "The number of electron shelld"
```
Comment (cjosey) [inline python/src/block/PhotoatomicComptonProfileBlock.python.cpp:48]: Extraneous `d`.

File: python/src/block/PhotoatomicElectronShellBlock.python.cpp [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+  // wrap views created by this block
+
+  // create the block
+  python::class_< Block > block(
+
+    module,
+    "PhotoatomicElectronShellBlock",
+    "The photoatomic LNEPS, LBEPS and LPIPS block with the electron shell data\n\n"
+    "The PhotoatomicPrincipalCrossSectionBlock class contains 3 arrays of the\n"
```
Comment (cjosey) [inline python/src/block/PhotoatomicElectronShellBlock.python.cpp:28]: Photoatomic Principal Cross Section Block is not correct here.

File: src/ACEtk/PhotoatomicTable.hpp [hunk 4]
```diff
@@ -17,23 +20,26 @@ namespace ACEtk {
...
+ *  The PhotoatomicTable class represents the ACE table for the photoatomic data.
  */
-class PhotoAtomicTable : protected Table {
+class PhotoatomicTable : protected Table {
 
   /* fields */
   block::ESZG eszg_;
   block::JINC jinc_;
   block::JCOH jcoh_;
+  block::JFLO jflo_;
```
Comment (cjosey) [inline src/ACEtk/PhotoatomicTable.hpp:31]: Technically, JFLO, EPS, and SWD are optional, but I'm not worried much about that.

Edit: I see, they're just length zero if the NXS values say they're not there.

File: src/ACEtk/PhotoatomicTable/src/generateBlocks.hpp [hunk 5]
```diff
@@ -17,7 +17,7 @@ auto block( std::size_t index ) const {
       end = *iter;
     }
   }
-
+std::cout << index << ' ' << start << ' ' << end << std::endl;
```
Comment (cjosey) [inline src/ACEtk/PhotoatomicTable/src/generateBlocks.hpp:20]: I suspect this is a leftover debug message.

**Review by cjosey — APPROVED** [review]

File: python/src/PhotoatomicTable.python.cpp (hunk 1 above)
Comment (whaeck) [reply python/src/PhotoatomicTable.python.cpp:63]: Yeah, that should be jflo

File: python/src/block/PhotoatomicComptonProfileBlock.python.cpp (hunk 2 above)
Comment (whaeck) [reply python/src/block/PhotoatomicComptonProfileBlock.python.cpp:48]: Fat fingers. That second d should have been an `s`

File: src/ACEtk/PhotoatomicTable.hpp (hunk 4 above)
Comment (whaeck) [reply src/ACEtk/PhotoatomicTable.hpp:31]: I've probably replaced these with optionals later anyway.

**Pull request merged**